pallets/anchors/src/lib.rs

```diff
@@ -356,6 +372,8 @@ pub mod pallet {
 
 				<PreCommitEvictionBuckets<T>>::remove((evict_bucket, idx));
 
+				<T as pallet::Config>::Currency::unreserve(&who, T::PreCommitDeposit::get());
```

Comment:
We cannot do this like this here, the `who` origin can be triggered by anyone, so we cannot give the deposit to the invoker. We need to get the identity from the PreCommit storage instead here, and unreserve on it.